### PR TITLE
feat: add support for bittwister

### PR DESCRIPTION
**Pull request by smuu**
## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: pkg/knuu/instance.go [hunk 1]
```diff
@@ -896,6 +975,70 @@ func (i *Instance) DisableNetwork() error {
 	return nil
 }
 
+// SetBandwidthLimit sets the bandwidth limit of the instance
+// bandwidth limit in bps (e.g. 1000 for 1Kbps)
+// Currently, only one of bandwidth, jitter, latency or packet loss can be set
```
Comment (mojtaba-esk) [inline pkg/knuu/instance.go:980]: Where this limitation is coming from?

File: pkg/knuu/instance.go [hunk 2]
```diff
@@ -896,6 +975,70 @@ func (i *Instance) DisableNetwork() error {
...
+// This function can only be called in the state 'Commited'
+func (i *Instance) SetBandwidthLimit(limit int) error {
+	if !i.IsInState(Committed) {
+		return fmt.Errorf("setting bandwidth limit is only allowed in state 'Committed'. Current state is '%s'", i.state.String())
+	}
+	if i.isNetworkConfigSet() {
+		return fmt.Errorf("setting bandwidth limit is not allowed if network config is already set")
+	}
+	i.networkConfig.bandwidth = limit
```
Comment (mojtaba-esk) [inline pkg/knuu/instance.go:989]: Ideally it should be a function call which calls a function from the bitTwister SDK.

File: pkg/knuu/instance_helper.go [hunk 3]
```diff
@@ -531,3 +570,43 @@ func (i *Instance) addOtelCollectorSidecar() error {
...
+	}
+	return networkConfigInstance, nil
+}
+
+func (i *Instance) addNetworkConfigSidecar() error {
+	networkConfigSidecar, err := i.createNetworkConfigInstance()
+	if err != nil {
+		return fmt.Errorf("error creating network config instance '%s': %w", i.k8sName, err)
+	}
+	bittwisterCmd := []string{"./bittwister", "start", "-d", "eth0"}
```
Comment (mojtaba-esk) [inline pkg/knuu/instance_helper.go:593]: This should not be hard coded. 
bittwister will be running as a service and knuu communicates through API calls via an SDK.

**Review by mojtaba-esk — COMMENTED** [review]
Left some comments, the proposal is to have an SDK for bittwister and we simply use it in knuu. I am working on it.

File: pkg/knuu/instance_helper.go (hunk 3 above)
Comment (smuu) [reply pkg/knuu/instance_helper.go:593]: This is temporary until the SDK with the API is ready.

Comment (mojtaba-esk) [reply pkg/knuu/instance_helper.go:593]: perfekt, let's give it a try and see how it goes while I am working on the API thing

**Review by mojtaba-esk — APPROVED** [review]

**Comment by smuu** [comment]
Closed if favor of https://github.com/celestiaorg/knuu/pull/210

**Pull request closed without merging**